### DISCUSSION TITLE
Can the rate be configured when using a bridge to forward messages?

### DISCUSSION THREAD

**Discussion by xigua88-66** [emqx/emqx#4702] [Q&A] [answered]
**Please describe your problem in detail, if necessary, you can upload the log file through the attachment**:
When I configure replayq_dir, the forwarding rate is about 4000msg/sec. If memory is used without configuration, the forwarding rate is 5000msg/sec. But when I increase the message sending rate, for example, when it reaches 10000msg/sec, the forwarding rate will drop to about 2000msc/sec. Can I configure the message forwarding rate to increase the message forwarding speed?

**Comment by hjianbo** [comment, marked as answer]
No, you can't.

Currently, there is still no configuration that can increase the rate of message forwarding. This is likely to be the bottleneck of what the system can handle.


How many totals cores and memory for your machine? And what's the size of the single message?

**Comment by xigua88-66** [comment]
I am using emqtt_bench for testing, and the message size uses the default value of 256 bytes. But when I try to increase the message size to 5120, the forwarding rate will drop to 1000msg/sec~2000msg/sec。
Are there other ways to increase the forwarding rate of bridged messages? 
Thanks for answering！
![image](https://user-images.githubusercontent.com/54090607/116186132-9b084000-a755-11eb-9395-ac0eefdd3fee.png)

**Comment by zmstone** [comment]
if you are forwarding messages from one EMQX node to another EMQX node
you can try the RPC bridge: https://docs.emqx.io/en/broker/v4.3/bridge/bridge-rpc.html#configure-broker-address-for-rpc-bridge

Otherwise I'm afraid you will have to wait for a MQTT connection pool implemented.
